opentelemetry/opentelemetry.go

```diff
@@ -0,0 +1,94 @@
+package opentelemetry
+
+import (
+	"context"
+	"encoding/hex"
+	"net/url"
+
+	"github.com/ChainSafe/chainbridge-core/config/relayer"
+	"github.com/ChainSafe/chainbridge-core/relayer/message"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/trace"
+	tracer "go.opentelemetry.io/otel/trace"
+)
+
+type Telemetry struct {
+	metrics *ChainbridgeMetrics
+	tracer  tracer.Tracer
+}
+
+// NewOpenTelemetry initializes OpenTelementry metrics and traces exporter to
+// OpenTelementry collector if set in relayer config or by default initializes
+// Prometheus on path and port setup by the relayer config
+func NewOpenTelemetry(config relayer.RelayerConfig) (*Telemetry, error) {
+	if config.OpenTelemetryCollectorURL != "" {
```

Comment:
More over there is no way to disable metrics

Reply:
My proposition is to make it with injecting telemetry implementation in to the relayer, so if nothing were injected then no metrics should be collected.
This also will makes it easier to implement different telemetry modules

Reply:
I kinda copied the existing implementation that had Prometheus by default, but 
I see it doesn't really make sense.

By injecting telemetry you mean something like this?

```
type Relayer {
   telemetry SomeTelemetryInterface
}

func NewRelayer( ........ , telemetry SomeTelemetryInterface) {
  ....
}
```

Should we separate tracer and metrics interfaces and just accept a Tracer interface and Metrics interface 
and expect them to always be present and if someone wants to turn off metrics he passes an dummy metrics implementation? 

The annoying thing I would like to avoid is having to check every time I want to trace something
or change metrics value to have to check if tracer or metrics are nil as 
it would bloat the actual implementation and if someone forgets to check it will easily panic.

Reply:
Also, would it make sense to make `Relayer` then private to avoid having to deal with such headaches if someone wants to avoid the constructor?